types/index.d.ts

```diff
@@ -86,5 +92,6 @@ export interface Chalk {
 	bgWhiteBright: Chalk;
 }
 
-declare function chalk (): any;
-export default chalk as Chalk;
+declare const chalk: Chalk & { supportsColor: ColorSupport };
```

Comment:
Why was `& { supportsColor: ColorSupport }` moved out of the `Chalk` interface? Or rather, what's the benefit of having it here?

Reply:
AFAICT `supportsColor` only exists on the default export? If it is left in the `Chalk` interface it will show up all over where it doesn't exist.

Reply:
You mean that this would be valid as well?

```js
chalk.red.supportsColor
```

I think you're right, it shouldn't.

Reply:
That's a good point and I agree, but shouldn't that apply to `constructor`, `enabled`, and `level` too then?

Reply:
I agree, but that's not what it looks like at runtime.

These things are possible:
`chalk.blue.enabled`
`chalk.blue.constructor`
`chalk.blue.level`

So if it shouldn't be possible, we will have to rewrite some internals so that the type definitions match what we have at runtime.

Reply:
@SamVerschueren 
>You mean that this would be valid as well?
 Yeah

or it could be changed to behave like `enabled` and `level`

Reply:
Hmm, there might be some use-case where people want to do:

```js
export const color = chalk.blue.bold;
```

```js
import {color} from '.';

color.enabled = false;
```

Reply:
@sindresorhus, good use-case -- and that is supported with these typings. But `color.supportsColor` would not be (nor does it exist at runtime).

should `supportsColor` be added to the chained/built chalk properties?

Reply:
Only if we add it in the runtime, which is a separate discussion to this PR. I'm undecided whether it makes sense though.